provider/oauth2/views.py

```diff
@@ -137,3 +145,35 @@ def invalidate_access_token(self, at):
         else:
             at.expires = now() - timedelta(days=1)
             at.save()
+
+
+class AccessTokenDetailView(View):
+    """
+    This view returns info about a given access token. If the token does not exist or is expired, HTTP 400 is returned.
+
+    A successful response has HTTP status 200 and includes a JSON object containing the username, scope, and expiry
+    time (in seconds) for the access token.
+
+    Example
+        GET /access_token/abc123/
+
+        {
+            username: "some-user",
+            scope: "read",
+            expires_in: 60
+        }
+    """
+
+    def get(self, request, *args, **kwargs):
+        JSON_CONTENT_TYPE = 'application/json'
+
+        try:
+            access_token = AccessToken.objects.get_token(kwargs['token'])
+            content = {
+                'username': access_token.user.username,
+                'scope': access_token.get_scope_display(),
+                'expires_in': access_token.get_expire_delta()
+            }
+            return HttpResponse(json.dumps(content), content_type=JSON_CONTENT_TYPE)
+        except ObjectDoesNotExist:
+            return HttpResponseBadRequest(json.dumps({'error': 'invalid_token'}), content_type=JSON_CONTENT_TYPE)
```

Comment:
Nah. The only other place where this value is used is in the test, and I'd prefer to share as little code as possible between tests and implementations. If the value does end up used elsewhere in the future, we will make a constant.